src/codeu/chat/client/commandline/Chat.java

```diff
@@ -329,6 +329,30 @@ public void invoke(List<String> args) {
         }
       }
     });
+    // I-REMOVE (remove interests)
+    //
+    // Add a command that will add a new interest when the user
+    // enters "i-add" while on the user panel.
+    //
+    panel.register("i-remove", new Panel.Command() {
+      @Override
+      public void invoke(List<String> args) {
+        try {
+          final Uuid id = args.size() > 0 ? Uuid.parse(args.get(0)) : Uuid.NULL;
+          if (args.size() > 0) {
+            if (id == null) {
+              System.out.println("ERROR: Failed to remove new interest");
```

Comment:
Strictly speaking, it's not that you couldn't remove a "new" interest, it's that no interest was specified, right?

Actually, is this statement reachable at all? You check `args.size() > 0` twice, once in the `?:` expression and once in the if-statement. Can you get to this line at all?

I see that you copied the structure from `i-add`, and indeed it looks weird to me there, too. I know @nvaa36 is on vacation, and I know I missed this on reviewing the `medium_feature_branch` the first time, but for what it's worth: It seems to me like a clearer structure would be

`if args are empty {`
`print "ERROR: Need to specify an interest`
`return`
`}`
`do everything else  // we know that args are not empty from here on out`

Does that seem clearer to you?